packages/xrpl-types/src/msg.rs

```diff
@@ -0,0 +1,423 @@
+        let s = match self {
+            XRPLMessageType::Proof => "proof",
+            XRPLMessageType::InterchainTransfer => "interchain_transfer",
+            XRPLMessageType::CallContract => "call_contract",
+            XRPLMessageType::AddGas => "add_gas",
+            XRPLMessageType::AddReserves => "add_reserves",
+        };
+        write!(f, "{}", s)
+    }
+}
+
+impl XRPLMessage {
+    pub fn tx_id(&self) -> HexTxHash {
+        match self {
+            XRPLMessage::ProverMessage(prover_message) => prover_message.tx_id.clone(),
+            XRPLMessage::InterchainTransferMessage(interchain_transfer_message) => {
+                interchain_transfer_message.tx_id.clone()
+            }
+            XRPLMessage::CallContractMessage(call_contract_message) => {
+                call_contract_message.tx_id.clone()
+            }
+            XRPLMessage::AddGasMessage(add_gas_message) => add_gas_message.tx_id.clone(),
+            XRPLMessage::AddReservesMessage(add_reserves_message) => {
+                add_reserves_message.tx_id.clone()
+            }
+        }
+    }
+
+    pub fn cc_id(&self, source_chain: ChainNameRaw) -> Option<CrossChainId> {
+        match self {
+            XRPLMessage::InterchainTransferMessage(interchain_transfer_message) => {
+                Some(interchain_transfer_message.cc_id(source_chain))
+            }
+            XRPLMessage::CallContractMessage(call_contract_message) => {
+                Some(call_contract_message.cc_id(source_chain))
+            }
+            XRPLMessage::ProverMessage(_) => None,
+            XRPLMessage::AddGasMessage(_) => None,
+            XRPLMessage::AddReservesMessage(_) => None,
+        }
+    }
+
+    pub fn with_cc_id(&self, source_chain: ChainNameRaw) -> WithCrossChainId<Self> {
+        WithCrossChainId::new(self.clone(), self.cc_id(source_chain))
+    }
+
+    pub fn hash(&self) -> [u8; 32] {
+        match self {
+            XRPLMessage::ProverMessage(prover_message) => prover_message.hash(),
+            XRPLMessage::InterchainTransferMessage(interchain_transfer_message) => {
+                interchain_transfer_message.hash()
+            }
+            XRPLMessage::CallContractMessage(call_contract_message) => call_contract_message.hash(),
+            XRPLMessage::AddGasMessage(add_gas_message) => add_gas_message.hash(),
+            XRPLMessage::AddReservesMessage(add_reserves_message) => add_reserves_message.hash(),
+        }
+    }
+}
+
+/// Represents an XRPL `Payment` transaction towards the XRPL multisig,
+/// performed by an XRPL user to initiate an interchain transfer call.
+/// Such messages are verified by the XRPL Voting Verifier and routed by the Router.
+#[cw_serde]
+#[derive(Eq, Hash)]
+pub struct XRPLInterchainTransferMessage {
+    #[serde(with = "hex_tx_hash")]
+    #[schemars(with = "String")] // necessary attribute in conjunction with #[serde(with ...)]
+    pub tx_id: HexTxHash,
```

Comment:
```suggestion
    pub msg_id: HexTxHash,
```

I would apply this change to all of the messages

Reply:
We're using `msg_id` in AddGas to refer to the TX hash of the message that's being topped up: https://github.com/axelarnetwork/axelar-amplifier/pull/764/commits/e5b88ef63558295648fd1c5d560dae083ef0a9c3

But we're still using `tx_id` to refer to the TX hash of the current message everywhere